tools/twix/src/panels/image/overlays/pose_detection.rs

```diff
@@ -42,69 +49,123 @@ impl Overlay for PoseDetection {
-
-            // draw keypoints
-            for keypoint in keypoints.iter() {
-                if keypoint.confidence < keypoint_confidence_threshold {
-                    continue;
-                }
-                painter.circle_stroke(keypoint.point, 5.0, Stroke::new(2.0, Color32::BLUE));
-
-                painter.floating_text(
-                    keypoint.point,
-                    Align2::LEFT_TOP,
-                    format!("{:.2}", keypoint.confidence),
-                    FontId::default(),
-                    Color32::WHITE,
-                );
-            }
-
-            // draw skeleton
-            for (idx1, idx2) in POSE_SKELETON {
-                let keypoint1 = &keypoints[idx1];
-                let keypoint2 = &keypoints[idx2];
-                if keypoint1.confidence < keypoint_confidence_threshold
-                    || keypoint2.confidence < keypoint_confidence_threshold
-                {
-                    continue;
-                };
-                painter.line_segment(
-                    keypoint1.point,
-                    keypoint2.point,
-                    Stroke::new(2.0, Color32::GREEN),
-                )
-            }
+        let unfiltered_human_poses: Vec<HumanPose> =
+            self.unfiltered_human_poses.require_latest()?;
+        let filtered_human_poses: Vec<HumanPose> = self.filtered_human_poses.require_latest()?;
+
+        if self.paint_unfiltered_poses {
+            paint_poses(
+                painter,
+                unfiltered_human_poses,
+                Color32::LIGHT_RED,
+                Color32::RED,
+                Color32::WHITE,
+                Align2::RIGHT_BOTTOM,
+            )?;
+        } else {
+            paint_poses(
+                painter,
+                filtered_human_poses,
+                Color32::BLUE,
+                Color32::DARK_BLUE,
+                Color32::WHITE,
+                Align2::RIGHT_BOTTOM,
+            )?;
         }
+
+        paint_detection_dead_zone(painter);
+
         Ok(())
     }
+
+    fn config_ui(&mut self, ui: &mut eframe::egui::Ui) {
+        ui.horizontal(|ui| {
+            ui.add_space(10.0);
+            ui.checkbox(&mut self.paint_unfiltered_poses, "Unfiltered Poses");
+        });
+    }
+}
+
+fn paint_detection_dead_zone(painter: &crate::twix_painter::TwixPainter<Pixel>) {
+    painter.rect_filled(
+        point![0.0, 0.0],
+        point![DETECTION_IMAGE_START_X, DETECTION_IMAGE_HEIGHT],
+        Color32::RED.gamma_multiply(0.3),
+    );
+    painter.rect_filled(
+        point![DETECTION_IMAGE_START_X + DETECTION_IMAGE_WIDTH, 0.0],
+        point![IMAGE_WIDTH, DETECTION_IMAGE_HEIGHT],
+        Color32::RED.gamma_multiply(0.3),
+    );
+}
+
+fn paint_poses(
+    painter: &crate::twix_painter::TwixPainter<Pixel>,
+    poses: Vec<HumanPose>,
+    line_color: Color32,
+    point_color: Color32,
+    text_color: Color32,
+    align: Align2,
```

Comment:
You always pass the same value for this parameter, consider removing it.

Reply:
Done